File: FinalProject/finalProject-js/js/diet.js
// Constructor
// AddDietItem
// EatFoodItem
// CanEat
// HasDiet
// ListDiet

class Diet {

    constructor(){
        this._dietItems = new Array();
    }

    AddDietItem(food, quantity){
        let tmpFood = new FoodItem(food, quantity);
        this._dietItems.push(tmpFood);
        alert(`Successfully added ${food} to the diet!`);
        
    }

    EatFoodItem(food, quantity, petName){
        if (this.Contains(food)){
            alert(`${petName} ate ${quantity} grams of ${food}!`)
        }
        else{
            alert(`${petName} doesn't eat ${food}.`)
        }
    }

    HasDiet(){
        if (this._dietItems.length > 0){
            return true;
        }
        else{
            return false;
        }
    }

    Contains(food){
        for (let i = 0; i < this._dietItems.length; i++){
            if (this._dietItems[i].Food() == food){
                // Returning ther food item for 
                return this._dietItems[i];
            }
        }
        return 0;
    }
}

class FoodItem {
    constructor(food, quantity){
        this._food = food;
        this._quantity = quantity;
    }

    Food(){
        return this._food;
    }
    Quantity(){
        return this._quantity;
    }
}

File: FinalProject/finalProject-cs/finalProject/Diet.cs
using System;
using System.Collections.Generic;

namespace finalProject
{
    public class Diet
    {
        private Dictionary<string, decimal> _diet = null;

        public Diet()
        {
            _diet = new Dictionary<string, decimal>();
        }

        public void AddDietItem(string food, decimal quantity)
        {
            _diet.Add(food, quantity);
            Console.WriteLine($"\n  Successfully added {food} to the diet!");
        }

        public void EatFoodItem(string food, string petName)
        {
            if (_diet.ContainsKey(food))
            {
                _diet.TryGetValue(food, out decimal qty);
                Console.WriteLine($"\n  {petName} ate {qty} grams of {food}!");
            }
            else
            {
                Console.WriteLine($"\n  {petName} doesn't eat {food}.");
            }
        }

        public bool CanEat(string food)
        {
            if (_diet.ContainsKey(food))
            {
                return true;
            }
            else
            {
                return false;
            }
        }

        public bool HasDiet()
        {
            if (_diet.Count > 0)
            {
                return true;
            }
            else
            {
                return false;
            }
        }

        public void ListDiet()
        {
            Console.WriteLine("  \n  - Diet Options -");
            foreach (string food in _diet.Keys)
            {
                decimal qty;
                _diet.TryGetValue(food, out qty);

                Console.WriteLine($"   {food}: {qty} grams");
            }
        }
    }
}
